src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -244,14 +253,20 @@ void shutdown() {
           Futures.catchingAsync(
               uploadResult,
               StatusRuntimeException.class,
-              (sre) -> Futures.immediateFailedFuture(new IOException(sre)),
+              (sre) -> Futures.immediateFailedFuture(
+                  new IOException(
+                      String.format(
+                          "Error while uploading artifact with digest '%s/%s'",
```

Comment:
It's an improvement, but it would be even better if it gave a filename.

Reply:
Absolutely! However, we won't get a filename here because the blob may not be a named file (e.g., `std{out,err}`). We would need to do that further downstream.

Let's start with the hash and I'll send a followup if I can figure out how to create an error message that contains the name.

Reply:
Maybe catch the exception and add filename at the call site. e.g. https://github.com/coeuvre/bazel/blob/remote-with-rx/src/main/java/com/google/devtools/build/lib/remote/GrpcCacheClient.java#L396